File: programmers/Lv_1/KimHyunseo/문자열내마음대로정렬하기.cpp
#include<iostream>
#include <string>
#include <vector>
#include <algorithm>
//저장해놓고
//
using namespace std;

vector<string> solution(vector<string> strings, int n) {
	vector<string> answer;
	vector<char> temp;
	vector<bool> overlap(strings.size(),true);
	for (int i = 0; i < strings.size(); i++)
	{
		temp.push_back(strings[i].at(n));
		cout << temp[i] << endl;
 	}
	sort(strings.begin(), strings.end());
	sort(temp.begin(), temp.end());
	//overlap부분만 정렬
	for (int i = 0; i < strings.size(); i++)
	{
		for (int j = 0; j < strings.size(); j++){
			if (overlap[j]) {
				if (temp[i] == strings[j].at(n))
				{
					answer.push_back(strings[j]);
					overlap[j] = false;
				}
			}
		}
	}
	
	return answer;
}

int main()
{
	vector<string> string = { "sun", "bed", "car" };
	                         
	solution(string, 1);
}

File: programmers/Lv_1/HwangJunghoon/문자열 내 마음대로 정렬하기.py
def solution(strings, n):
    return sorted(sorted(strings), key=lambda strings: strings[n])
# 주어진 인덱스 정렬 이외에는 사전순 배열을 해야 하므로 미리 sort를 해놓고 람다를 사용해서 인덱스로 정렬해야 함
# 그냥 하면 예시2에서 abce가 abcd보다 먼저 오게 됨